Send local pulumi traces to Jaeger

twunderlich-grapl opened the pull request:
<!-- Thank you for your contribution to Grapl! Please take some time
to fill out this short template to help us review and understand your
PR. -->

### Which issue does this PR correspond to?
https://github.com/grapl-security/issue-tracker/issues/862
<!-- Please provide a link to the GitHub Issue this pull request is
meant to address. -->

### What changes does this PR make to Grapl? Why?
This sends pulumi's traces to Jaeger when running local grapl.

Note that this relies on https://github.com/grapl-security/grapl/pull/1508 being merged in first
<!-- Please describe the functional impact of these changes and the
rationale behind them. This will help us understand your approach to
solving the problem. -->

### How were these changes tested?
`make up` on a branch with the observability job merged in.
Go to http://localhost:16686/search, you should see the pulumi traces. 
<!-- Please describe how these changes were tested. There should be
sufficient detail that reviewers can replicate your testing
procedure. If the procedure is a manual one that's OK, your
description will help us work with you to get it automated. -->

Commit:
Send pulumi traces to Jaeger

For local grapl, we should send all pulumi traces to Jaeger. Since
the containerized pulumi is only used in local grapl, this should just
work

christophermaier commented on docker-compose.yml:
```diff
@@ -31,7 +31,7 @@ services:
           # login --local doesn't support organizations
           pulumi stack init local-grapl-integration-tests --non-interactive
-          pulumi up --yes --skip-preview --stack=local-grapl-integration-tests
+          pulumi up --yes --skip-preview --stack=local-grapl-integration-tests --tracing http://host.docker.internal:9411/api/v1/spans
```
Since this command is basically one big chunk of bash script, can we set this URL as a variable up at the top and just reference it in both these places?

christophermaier reviewed (approved):
![](https://media.giphy.com/media/NSDmc1yecieZBbMSFG/giphy.gif)

Commit:
Use a variable for the trace URL

We have to use the double dollar sign ($$) since docker-compose
interprets variables with a single dollar sign as environment variables.

christophermaier commented on docker-compose.yml:
```diff
@@ -14,11 +14,12 @@ services:
     command:
       - |
         pulumi login --local
+        export TRACING_BACKEND_URL="http://host.docker.internal:9411/api/v1/spans"
```
Tiny nitpick: I don't think `export` is required here; the variable doesn't need to be available to subprocesses, since you're passing it directly to the process via the command line option.

Commit:
Remove unnecessary export
